nucypher/network/nodes.py

```diff
@@ -610,7 +610,7 @@ def block_until_number_of_known_nodes_is(self,
             elapsed = (round_finish - start).seconds
             if elapsed > timeout:
                 if len(self.known_nodes) >= number_of_nodes_to_know:  # Last chance!
-                    continue
+                    return True
```

Comment:
May be worth adding a 
```
self.log.info("Learned about enough nodes after {} rounds.".format(rounds_undertaken))
```
to match the exit branch at the beginning of the `while` loop (in other words, so that we logged something on success regardless of the codepath that lead to it).

Reply:
✔️